### DISCUSSION TITLE
How to get more information about failed join attempts?

### DISCUSSION THREAD

**Discussion by tamcore** [gravitational/teleport#28990] [Q&A]
Hi,

in one of our Teleport clusters, we've noticed a ton of log entries like
```
2023-07-11T05:39:39Z WARN [AUTH]      "95853bcb-2f7c-4b68-bb51-7a08b7939eda" can not join the cluster with role Node, token error: token expired or not found auth/join.go:52
2023-07-11T05:39:39Z INFO [AUTH]      Node "95853bcb-2f7c-4b68-bb51-7a08b7939eda" [4ccc6f4f-1242-4114-b29d-aa28794ed475] is trying to join with role: Node. auth/join.go:106
```
about agents not being able to join. It's only about 4 different agents, so not a lot of traffic. But we simply can't find any information about those nodes.

Is there any way, to produce logs with the IP and / or attempted token of said agent, so we can find out who exactly is the origin of that traffic? Increasing log severity to debug didn't help.

**Comment by webvictim** [comment]
Unfortunately Teleport doesn't currently output the IP or anyother details about the join attempt. You could try `tctl get node/95853bcb-2f7c-4b68-bb51-7a08b7939eda` or `tctl get node/4ccc6f4f-1242-4114-b29d-aa28794ed47` but they're unlikely to work as the node isn't actually registered yet.

Another way might be to use `ss` or `iptables` to monitor/watch incoming connections. Obviously in a high traffic cluster this would be hard, but you might be able to isolate down the IPs against nodes that you know of.

If you're running the OSS version, you could also look at forking the code to add additional log output about the source IP for join attempts.

One thing that's interesting to me is that on v13, when I try to join a node with the hostname `testnode` I actually see that string in the logs:

```
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z INFO [AUTH]      Node "testnode" [bf86658e-4e81-4d77-970f-ff4c12938ad2] is trying to join with role: Node. auth/join.go:106
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z WARN [AUTH]      "testnode" can not join the cluster with role Node, token error: token expired or not found auth/join.go:52
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z DEBU [AUTH]      ClientCertPool -> cert(purple issued by purple:10094116141712434593203685919031113190) auth/middleware.go:768
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z DEBU [AUTH]      ClientCertPool -> cert(purple issued by purple:162071946948227685076371702976342367271) auth/middleware.go:768
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z INFO [AUTH]      Node "testnode" [bf86658e-4e81-4d77-970f-ff4c12938ad2] is trying to join with role: Instance. auth/join.go:106
Jul 12 12:02:14 ip-172-31-30-140 teleport[2438539]: 2023-07-12T12:02:14Z WARN [AUTH]      "testnode" can not join the cluster with role Instance, token error: token expired or not found auth/join.go:52
```

Is it possible that your hostnames are all just UUIDs, or are you running an older version of Teleport?
